wallet/wallet.go

```diff
@@ -0,0 +1,35 @@
+package wallet
+
+import "errors"
+
+type Bitcoin float64
+
+type Wallet struct {
+	balance Bitcoin
+}
+
+func (w *Wallet) Deposit(amount Bitcoin) error {
+	if amount < 0 {
+		return ErrNegativeInput
+	}
+	w.balance += amount
+	return nil
+}
+
+var ErrInsufficientFunds = errors.New("cannot withdraw, insufficient funds")
+var ErrNegativeInput = errors.New("negative inputs are not allowed")
```

Comment:
previously you had only `ErrInsufficientFunds`, so it was good having it close to `Withdraw()` (the only place where it's used). But now you have 2 errors, that are used in different places, so let's move them to the top of the file